File: 神龙三厂胎压设备搬迁项目VS/CCar.vb
Option Strict Off
Option Explicit On
Public Class CCar
    Inherits DSGTest.Common.Comm.CCar

    Private LastVin As String
    Public printFlag As Boolean
    Public LastCar As CCar

    Public Sub Save()
        On Error GoTo CCAR_SAVE_ERR
        Dim cnn As New ADODB.Connection
        Dim rs As New ADODB.Recordset
        Dim lastrs As ADODB.Recordset
        Dim sql As String
        Dim mtoc As String
        cnn.CommandTimeout = TimeOutNum
        cnn.Open(DBCnnStr)

        lastrs = cnn.Execute("select ""VIN"" from ""T_Result"" order by ""ID"" desc limit 1")
        printFlag = False
        If Not lastrs.EOF Then
            LastVin = lastrs.Fields(0).Value
        End If
        LastCar = New CCar
        If LastVin <> m_VINCode And LastVin <> "" Then

            LastCar.SetCarInfByVIN(LastVin)
            If m_TireRFID <> "" And m_TireRFID <> "00000000" Then
                Select Case m_TireRFID
                    Case LastCar.TireLFID
                        printFlag = True
                        LastCar.TireLFID = "00000000"
                        m_TireRFID = "00000000"
                    Case LastCar.TireLRID
                        printFlag = True
                        LastCar.TireLRID = "00000000"
                        m_TireRFID = "00000000"
                    Case LastCar.TireRFID
                        printFlag = True
                        LastCar.TireRFID = "00000000"
                        m_TireRFID = "00000000"
                    Case LastCar.TireRRID
                        printFlag = True
                        LastCar.TireRRID = "00000000"
                        m_TireRFID = "00000000"
                End Select
            End If

            If m_TireLFID <> "" And m_TireLFID <> "00000000" Then
                Select Case m_TireLFID
                    Case LastCar.TireLFID
                        printFlag = True
                        LastCar.TireLFID = "00000000"
                        m_TireLFID = "00000000"
                    Case LastCar.TireLRID
                        printFlag = True
                        LastCar.TireLRID = "00000000"
                        m_TireLFID = "00000000"
                    Case LastCar.TireRFID
                        printFlag = True
                        LastCar.TireRFID = "00000000"
                        m_TireLFID = "00000000"
                    Case LastCar.TireRRID
                        printFlag = True
                        LastCar.TireRRID = "00000000"
                        m_TireLFID = "00000000"
                End Select
            End If

            If m_TireRRID <> "" And m_TireRRID <> "00000000" Then
                Select Case m_TireRRID
                    Case LastCar.TireLFID
                        printFlag = True
                        LastCar.TireLFID = "00000000"
                        m_TireRRID = "00000000"
                    Case LastCar.TireLRID
                        printFlag = True
                        LastCar.TireLRID = "00000000"
                        m_TireRRID = "00000000"
                    Case LastCar.TireRFID
                        printFlag = True
                        LastCar.TireRFID = "00000000"
                        m_TireRRID = "00000000"
                    Case LastCar.TireRRID
                        printFlag = True
                        LastCar.TireRRID = "00000000"
                        m_TireRRID = "00000000"
                End Select
            End If
            If m_TireLRID <> "" And m_TireLRID <> "00000000" Then
                Select Case m_TireLRID
                    Case LastCar.TireLFID
                        printFlag = True
                        LastCar.TireLFID = "00000000"
                        m_TireLRID = "00000000"
                    Case LastCar.TireLRID
                        printFlag = True
                        LastCar.TireLRID = "00000000"
                        m_TireLRID = "00000000"
                    Case LastCar.TireRFID
                        printFlag = True
                        LastCar.TireRFID = "00000000"
                        m_TireLRID = "00000000"
                    Case LastCar.TireRRID
                        printFlag = True
                        LastCar.TireRRID = "00000000"
                        m_TireLRID = "00000000"
                End Select
            End If
            'LastCar.Save
        End If


        testState = 0
        If Len(m_TireRFID) = 8 And m_TireRFID <> "00000000" And Trim(m_TireRFID) <> "" And m_TireRFID <> m_TireLFID And m_TireRFID <> m_TireLRID And m_TireRFID <> m_TireRRID Then
            testState = testState + 8
        End If
        If Len(m_TireLFID) = 8 And m_TireLFID <> "00000000" And Trim(m_TireLFID) <> "" And m_TireLFID <> m_TireRFID And m_TireLFID <> m_TireLRID And m_TireLFID <> m_TireRRID Then
            testState = testState + 4
        End If
        If Len(m_TireRRID) = 8 And m_TireRRID <> "00000000" And Trim(m_TireRRID) <> "" And m_TireRRID <> m_TireLFID And m_TireRRID <> m_TireLRID And m_TireRRID <> m_TireRFID Then
            testState = testState + 2
        End If
        If Len(m_TireLRID) = 8 And m_TireLRID <> "00000000" And Trim(m_TireLRID) <> "" And m_TireLRID <> m_TireLFID And m_TireLRID <> m_TireRFID And m_TireLRID <> m_TireRRID Then
            testState = testState + 1
        End If

        rs.Open("select * from ""T_Result"" where ""VIN""='" & m_VINCode & "' ", DBCnnStr, 1, 3)
        If rs.EOF Then
            rs.AddNew()
        End If


        rs.Fields("VIN").Value = m_VINCode
        rs.Fields("VIS").Value = Right(m_VINCode, 8)
        rs.Fields("ID020").Value = m_TireRFID
        rs.Fields("ID022").Value = m_TireLFID
        rs.Fields("ID021").Value = m_TireRRID
        rs.Fields("ID023").Value = m_TireLRID

        rs.Fields("Mdl020").Value = m_TireRFMdl
        rs.Fields("Mdl022").Value = m_TireLFMdl
        rs.Fields("Mdl021").Value = m_TireRRMdl
        rs.Fields("Mdl023").Value = m_TireLRMdl

        rs.Fields("Pre020").Value = m_TireRFPre
        rs.Fields("Pre022").Value = m_TireLFPre
        rs.Fields("Pre021").Value = m_TireRRPre
        rs.Fields("Pre023").Value = m_TireLRPre

        rs.Fields("Temp020").Value = m_TireRFTemp
        rs.Fields("Temp022").Value = m_TireLFTemp
        rs.Fields("Temp021").Value = m_TireRRTemp
        rs.Fields("Temp023").Value = m_TireLRTemp

        rs.Fields("Battery020").Value = m_TireRFBattery
        rs.Fields("Battery022").Value = m_TireLFBattery
        rs.Fields("Battery021").Value = m_TireRRBattery
        rs.Fields("Battery023").Value = m_TireLRBattery

        rs.Fields("AcSpeed020").Value = m_TireRFAcSpeed
        rs.Fields("AcSpeed022").Value = m_TireLFAcSpeed
        rs.Fields("AcSpeed021").Value = m_TireRRAcSpeed
        rs.Fields("AcSpeed023").Value = m_TireLRAcSpeed

        rs.Fields("TestTime").Value = Now
        rs.Fields("TestState").Value = testState
        rs.Fields("Dev").Value = "101"
        rs.Fields("UploadSign").Value = False
        rs.Fields("DownloadSign").Value = False

        '获取MTOC码
        mtoc = GetMtocFromVinColl(m_VINCode)
        If mtoc <> "" Then
            rs.Fields("MTOC").Value = mtoc
        End If

        rs.Update()
        rs.Close()
        'UPGRADE_NOTE: 在对对象 rs 进行垃圾回收前，不可以将其销毁。 单击以获得更多信息:“ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"”
        rs = Nothing
        cnn.Close()

        '存入远程数据库
        cnn.ConnectionTimeout = TimeOutNum 'Add by ZCJ 2012/02/21
        cnn.CommandTimeout = TimeOutNum
        cnn.Open(RDBCnnStr)

        rs.Open("select * from ""T_Result"" where ""VIN""='" & m_VINCode & "' ", RDBCnnStr, 1, 3)

        If rs.EOF Then
            rs.AddNew()
        End If


        rs.Fields("VIN").Value = m_VINCode
        rs.Fields("VIS").Value = Right(m_VINCode, 8)
        rs.Fields("ID020").Value = m_TireRFID
        rs.Fields("ID022").Value = m_TireLFID
        rs.Fields("ID021").Value = m_TireRRID
        rs.Fields("ID023").Value = m_TireLRID

        rs.Fields("Mdl020").Value = m_TireRFMdl
        rs.Fields("Mdl022").Value = m_TireLFMdl
        rs.Fields("Mdl021").Value = m_TireRRMdl
        rs.Fields("Mdl023").Value = m_TireLRMdl

        rs.Fields("Pre020").Value = m_TireRFPre
        rs.Fields("Pre022").Value = m_TireLFPre
        rs.Fields("Pre021").Value = m_TireRRPre
        rs.Fields("Pre023").Value = m_TireLRPre

        rs.Fields("Temp020").Value = m_TireRFTemp
        rs.Fields("Temp022").Value = m_TireLFTemp
        rs.Fields("Temp021").Value = m_TireRRTemp
        rs.Fields("Temp023").Value = m_TireLRTemp

        rs.Fields("Battery020").Value = m_TireRFBattery
        rs.Fields("Battery022").Value = m_TireLFBattery
        rs.Fields("Battery021").Value = m_TireRRBattery
        rs.Fields("Battery023").Value = m_TireLRBattery

        rs.Fields("AcSpeed020").Value = m_TireRFAcSpeed
        rs.Fields("AcSpeed022").Value = m_TireLFAcSpeed
        rs.Fields("AcSpeed021").Value = m_TireRRAcSpeed
        rs.Fields("AcSpeed023").Value = m_TireLRAcSpeed

        rs.Fields("Dev").Value = "101"
        rs.Fields("TestTime").Value = Now
        rs.Fields("TestState").Value = testState
        rs.Fields("UploadSign").Value = False
        rs.Fields("DownloadSign").Value = False
        If mtoc <> "" Then
            rs.Fields("MTOC").Value = mtoc
        End If

        rs.Update()
        rs.Close()
        'UPGRADE_NOTE: 在对对象 rs 进行垃圾回收前，不可以将其销毁。 单击以获得更多信息:“ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"”
        rs = Nothing
        cnn.Close()

        'UPGRADE_NOTE: 在对对象 cnn 进行垃圾回收前，不可以将其销毁。 单击以获得更多信息:“ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"”
        cnn = Nothing
        Exit Sub
CCAR_SAVE_ERR:
        LogWritter("CCAR_SAVE_ERR 错误信息：" & Err.Description)
    End Sub

    Public Function GetMtocFromVinColl(ByRef vin As String) As String
        On Error GoTo Err_Renamed
        Dim cnn As New ADODB.Connection
        Dim rs As New ADODB.Recordset
        cnn.CommandTimeout = TimeOutNum
        cnn.Open(DBCnnStr)
        rs = cnn.Execute("select mtoc from vincoll where vin='" & vin & "'")
        If Not rs.EOF Then
            GetMtocFromVinColl = rs.Fields("mtoc").Value & ""
        Else
            GetMtocFromVinColl = ""
        End If
        rs.Close()
        'UPGRADE_NOTE: 在对对象 rs 进行垃圾回收前，不可以将其销毁。 单击以获得更多信息:“ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"”
        rs = Nothing
        cnn.Close()
        Exit Function
Err_Renamed:
        GetMtocFromVinColl = ""
    End Function
End Class

File: DSGTest.Common/Comm/CCar.cs
using DSGTest.Common.SqlServive.Sql_DPCAWH1_DSG101;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSGTest.Common.Comm
{
    public class CCar
    {
        #region 属性
        protected string m_VINCode = string.Empty;
        public string VINCode { get => m_VINCode; set => m_VINCode = value; }

        protected string m_TireRFID = string.Empty;
        public string TireRFID { get => m_TireRFID; set => m_TireRFID = value; }

        protected string m_TireRFMdl = string.Empty;
        public string TireRFMdl { get => m_TireRFMdl; set => m_TireRFMdl = value; }

        protected string m_TireRFPre = string.Empty;
        public string TireRFPre { get => m_TireRFPre; set => m_TireRFPre = value; }

        protected string m_TireRFTemp = string.Empty;
        public string TireRFTemp { get => m_TireRFTemp; set => m_TireRFTemp = value; }

        protected string m_TireRFBattery = string.Empty;
        public string TireRFBattery { get => m_TireRFBattery; set => m_TireRFBattery = value; }

        protected string m_TireRFAcSpeed = string.Empty;
        public string TireRFAcSpeed { get => m_TireRFAcSpeed; set => m_TireRFAcSpeed = value; }

        protected string m_TireRRID = string.Empty;
        public string TireRRID { get => m_TireRRID; set => m_TireRRID = value; }

        protected string m_TireRRMdl = string.Empty;
        public string TireRRMdl { get => m_TireRRMdl; set => m_TireRRMdl = value; }

        protected string m_TireRRPre = string.Empty;
        public string TireRRPre { get => m_TireRRPre; set => m_TireRRPre = value; }

        protected string m_TireRRTemp = string.Empty;
        public string TireRRTemp { get => m_TireRRTemp; set => m_TireRRTemp = value; }

        protected string m_TireRRBattery = string.Empty;
        public string TireRRBattery { get => m_TireRRBattery; set => m_TireRRBattery = value; }

        protected string m_TireRRAcSpeed = string.Empty;
        public string TireRRAcSpeed { get => m_TireRRAcSpeed; set => m_TireRRAcSpeed = value; }

        protected string m_TireLFID = string.Empty;
        public string TireLFID { get => m_TireLFID; set => m_TireLFID = value; }

        protected string m_TireLFMdl = string.Empty;
        public string TireLFMdl { get => m_TireLFMdl; set => m_TireLFMdl = value; }

        protected string m_TireLFPre = string.Empty;
        public string TireLFPre { get => m_TireLFPre; set => m_TireLFPre = value; }

        protected string m_TireLFTemp = string.Empty;
        public string TireLFTemp { get => m_TireLFTemp; set => m_TireLFTemp = value; }

        protected string m_TireLFBattery = string.Empty;
        public string TireLFBattery { get => m_TireLFBattery; set => m_TireLFBattery = value; }

        protected string m_TireLFAcSpeed = string.Empty;
        public string TireLFAcSpeed { get => m_TireLFAcSpeed; set => m_TireLFAcSpeed = value; }

        protected string m_TireLRID = string.Empty;
        public string TireLRID { get => m_TireLRID; set => m_TireLRID = value; }

        protected string m_TireLRMdl = string.Empty;
        public string TireLRMdl { get => m_TireLRMdl; set => m_TireLRMdl = value; }

        protected string m_TireLRPre = string.Empty;
        public string TireLRPre { get => m_TireLRPre; set => m_TireLRPre = value; }

        protected string m_TireLRTemp = string.Empty;
        public string TireLRTemp { get => m_TireLRTemp; set => m_TireLRTemp = value; }

        protected string m_TireLRBattery = string.Empty;
        public string TireLRBattery { get => m_TireLRBattery; set => m_TireLRBattery = value; }

        protected string m_TireLRAcSpeed = string.Empty;
        public string TireLRAcSpeed { get => m_TireLRAcSpeed; set => m_TireLRAcSpeed = value; }
        #endregion

        protected short testState;

        public string GetTestState()
        {
            return testState.ToString();
        }

        public void SetCarInfByVIN(string vin)
        {
            var result = ServiceT_Result.Queryable(vin);
            if (result == null)
                return;
            m_VINCode = result.VIN;
            m_TireRFID = result.ID020;
            m_TireLFID = result.ID022;
            m_TireRRID = result.ID021;
            m_TireLRID = result.ID023;
            m_TireRFMdl = result.Mdl020;
            m_TireLFMdl = result.Mdl022;
            m_TireRRMdl = result.Mdl021;
            m_TireLRMdl = result.Mdl023;
            m_TireRFPre = result.Pre020;
            m_TireLFPre = result.Pre022;
            m_TireRRPre = result.Pre021;
            m_TireLRPre = result.Pre023;
            m_TireRFTemp = result.Temp020;
            m_TireLFTemp = result.Temp022;
            m_TireRRTemp = result.Temp021;
            m_TireLRTemp = result.Temp023;
            m_TireRFBattery = result.Battery020;
            m_TireLFBattery = result.Battery022;
            m_TireRRBattery = result.Battery021;
            m_TireLRBattery = result.Battery023;
            m_TireRFAcSpeed = result.AcSpeed020;
            m_TireLFAcSpeed = result.AcSpeed022;
            m_TireRRAcSpeed = result.AcSpeed021;
            m_TireLRAcSpeed = result.AcSpeed023;
            short.TryParse(result.TestState, out testState);
        }
    }
}
